### DISCUSSION TITLE
How to migrate nuclei templates from the old syntax to the new syntax

### DISCUSSION THREAD

**Discussion by Caratpine** [projectdiscovery/nuclei#5208] [Q&A] [answered]
### Discussed in https://github.com/projectdiscovery/nuclei/discussions/5185

<div type='discussions-op-text'>

<sup>Originally posted by **Caratpine** May 15, 2024</sup>
Hi masters.
We have made a template management ourselves, with about 2,000 templates. The nuclei version we use is 2.8.8. Recently we want to upgrade nuclei to the latest version, but because the templates we maintain are not new syntax, we need to migrate it. , are there any good solutions or tools? Thank you.</div>

**Comment by tarunKoyalwar** [comment, marked as answer]
@Caratpine , all breaking changes related to v3 are documented here https://github.com/projectdiscovery/nuclei/discussions/3648 

also Nuclei v3 warns about these deprecated syntax but support for these is not removed yet , so most of the templates should work with v3 with warnings and it is also recommended to update to most recent syntax and there aren't many breaking changes in v3 template wise 

template wise there is only 1 major change i.e change in protocol name ex: `requests: -> http:` and this can easily be automated via sed or any bash match replace

---

FYI, Nuclei doesn't have any LTS or Long term support for a particular version , this is because we consider Nuclei + Nuclei Templates as a single unit and not different entities , and we keep adding new features in Nuclei to meet immediate demand of a template[s]

Example: if a certain high/critical CVE requires a helper function or something else that is missing in latest version of nuclei , we try to add it and immediately create a new release along with nuclei-templates release for that. that way users always have latest and greatest  and are able to find / remmediate the vulnerability

**Comment by Caratpine** [comment]
Hi @tarunKoyalwar thanks for you reply, yes, i've checked the change log, and found the solution, thanks!
